spec/Asset-Exchange-Protocol.md

```diff
@@ -60,4 +60,12 @@ And the format of transfer transaction is as shown below.
+As described above in the 10-gold-to-100-silver order, it is possible to make a transfer transaction which has a 10-gold input that results in a 5-gold and 50-silver output, tagged with the same order. (Other inputs/outputs should be provided by a taker or a relayer). After this transaction, an asset contains the hash of the *consumed order*, which is 5-gold-to-50-silver order, not 10-gold-to-100-silver order. In order to use the 5-gold output, provide the 5-gold-to-50-silver order with the same information except for the `assetQuantityFrom` and the `assetQuantityTo` field. Neither a lock script nor an unlock script is needed for the 5-gold input. CodeChain will compare the order of an input and the order of the corresponding previous output, and will not run VM on the order if those orders are identical.
+
+## How to support cancellation?
+
+An order can be cancelled by the maker before fully filled. Let's use the described 10-gold-to-100-silver order example again. Suppose the maker wants to cancel the order after the 5-gold and 50-silver transaction is processed. Unlike the previous partial fill case, this time the maker has to prove the ownership of remained asset. Both a lock script and an unlock script is needed for the 5-gold input. Writing a new transaction with empty `orders` field with this input make Codechain to cancel the remained 5-gold and 50-silver order.
+
+## How to handle matched two orders with different exchange ratios?
+
+Suppose Alice and Bob want to exchange their own assets through a DEX platform. Alice offered 10-gold-to-100-silver exchange ratio but Bob offered 5-gold-to-100-silver exchange ratio which has higher gold to silver ratio compared to the Alice's offer. For DEX platforms to satisfy both Alice and Bob, the minimum conditions are "Alice gets at least 100 silvers at the expense of 5 golds" and "Bob gets at least 5 golds at the expense of 100 silvers". Satisfying this conditions, there are extra 5 golds. Because Codechain checks the minimum conditons only, DEX platforms have a freedom of how they dispose these extra assets. As long as the minimum conditions are satisfied, any possible choice of transaction could be accepted.
```

Comment:
Suppose Alice and Bob want to exchange their own assets through a DEX platform. Alice offered a 10-gold-to-100-silver exchange ratio but Bob offered a 5-gold-to-100-silver exchange ratio, which has higher gold to silver ratio compared to the Alice's offer. For DEX platforms to satisfy both Alice and Bob, the minimum conditions are "Alice gets at least 100 silvers at the expense of 5 golds" and "Bob gets at least 5 golds at the expense of 100 silvers". Satisfying these conditions, there are extra 5 golds. Because CodeChain only checks the minimum conditions, DEX platforms have a freedom of how they dispose these extra assets. As long as the minimum conditions are satisfied, any possible choice of transaction could be accepted.

Reply:
Thank you